### PR TITLE
Rails::Auth::ConfigBuilder

### PR DESCRIPTION
**Pull request by tarcieri**
Helper functions which build middleware chains for different environments.

Provides "convention over configuration" when using Rails::Auth inside a Rails app.

### CODE REVIEW

File: lib/rails/auth/config_builder.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      extend self
+
+      # Application-level configuration (i.e. config/application.rb)
+      def application(config, acl_file: Rails.root.join("config/acl.yaml"), matchers: ACL::DEFAULT_MATCHERS)
+        config.x.rails_auth.acl = Rails::Auth::ACL.from_yaml(
+          File.read(acl_file.to_s),
+          matchers: matchers
+        )
+
+        config.middleware.use Rails::Auth::ACL::Middleware, acl: config.x.acl
```
Comment (hablutzel1) [inline lib/rails/auth/config_builder.rb]: Shouldn't `config.x.rails_auth.acl = Rails::Auth::ACL.from_yaml` be `config.x.acl = Rails::Auth::ACL.from_yaml` (without the `rails_auth`) as later you are doing `config.middleware.use Rails::Auth::ACL::Middleware, acl: config.x.acl`?.

PS: Total newbie to Ruby and Rails here.

Comment (tarcieri) [reply lib/rails/auth/config_builder.rb]: At first glance you're right, and things are still this way on `master`, but if that's the case I'm not sure how anything is working.

Comment (hablutzel1) [reply lib/rails/auth/config_builder.rb]: Actually, it is not working (at least for me) in Rails 5.2.0 with something like this in application.rb:

```
Rails::Auth::ConfigBuilder.application(config, matchers: { allow_x509_subject: Rails::Auth::X509::Matcher })
```

So I had to resort to manual configuration, copying and modifying the body of the 'application' method.

Comment (tarcieri) [reply lib/rails/auth/config_builder.rb]: That definitely looks like a bug. Care to open an issue about it?

Comment (hablutzel1) [reply lib/rails/auth/config_builder.rb]: Done, https://github.com/square/rails-auth/issues/49.

**Pull request merged**